codeface_extraction/extractions.py

```diff
@@ -35,14 +35,21 @@
 #
 
 
-def get_extractions(dbm, conf, resdir, csv_writer, extract_impl, extract_on_range_level):
+def get_extractions(dbm, conf, resdir, csv_writer, extract_commit_messages, extract_impl, extract_on_range_level):
     # all extractions are sublcasses of Extraction:
     # instantiate them all!
     __extractions = []
     for cls in Extraction.__subclasses__():
-        if (extract_impl or
-                (str(cls) != "<class 'codeface_extraction.extractions.FunctionImplementationExtraction'>" and
-                 str(cls) != "<class 'codeface_extraction.extractions.FunctionImplementationRangeExtraction'>")):
+        if ((extract_impl and
```

Comment:
Unfortunately, this condition is quite elongated and confusing now. Can we do something like this instead:
```Python
# list of extractions to skip
extractions_to_skip = []
if not extract_impl:
    extractions_to_skip += ["<class 'codeface_extraction.extractions.CommitMessageExtraction'>"]
# add extractions to skip also for other flags

# collect all extractions and instantiate objects
for cls in Extraction.__subclasses__():
    __extractions.append(cls(dbm, conf, resdir, csv_writer))
# remove all extractions that are listed in extractions_to_skip
__extractions = filter(lambda x: str(x) not in extractions_to_skip, __extractions)
```

Maybe, we can also find a better way than the call to `str()` on the way.

------------------
Disclaimer: The code is untested as I do not have a proper Python installation on this computer.

Reply:
I also dislike the elongated if-statement, but it was a simple and easy way to introduce configuration parameter of the additional extraction. Nevertheless, your suggestion looks good to me (except for mixing implementation extraction and commit message extraction). I will try out your code suggestion tomorrow.

Reply:
Just one comment: The last line of your suggestion does not not work as we do not operate on classes any more, but on already instantiated classes, that is, objects. Nonetheless, I have just removed those line and added a simple if-statement before the `__extractions.append` statement. Please see my new commit.

Reply:
Thank you for this nice solution, @bockthom and @clhunsen.